java/ql/src/Security/CWE/CWE-094/InsecureBeanValidation.ql

```diff
@@ -14,6 +14,38 @@ import semmle.code.java.dataflow.TaintTracking
 import semmle.code.java.dataflow.FlowSources
 import DataFlow::PathGraph
 
+class ELMessageInterpolatorType extends RefType {
+  ELMessageInterpolatorType() {
+    this
+        .getASourceSupertype*()
+        .hasQualifiedName("org.hibernate.validator.messageinterpolation",
+          ["ResourceBundleMessageInterpolator", "ValueFormatterMessageInterpolator"])
+  }
+}
+
+class SetSafeMessageInterpolator extends MethodAccess {
+  SetSafeMessageInterpolator() {
+    exists(Method m |
+      this.getMethod() = m and
+      (
+        m
+            .getDeclaringType()
+            .getASourceSupertype*()
```

Comment:
Code style: You can pull the type declaration into its own variable, to share between the two cases.
```
exists(Method m, RefType t |
  this.getMethod() = m and
  m.getDeclaringType().getASourceSupertype*() = t and
  (
    t.hasQualifiedName(...) and m.hasName(...)
    or
    t.hasQualifiedName(...) and m.hasName(...)
  )
) and
...
```